packages/coinbase/src/index.ts

```diff
@@ -62,9 +89,20 @@ function coinbaseWallet({
 
           return coinbaseWalletProvider
         }
+        const provider = createEIP1193Provider(coinbaseWalletProvider, {
```

Comment:
@nateReiners patched provider to handle number value returned from `eth_chainId` - just flagging for visibility 😄

Reply:
@Adamj1232 If you want to bump the sdk to 4.0.3 and get rid of this workaround, the version with the `eth_chainId` fix was just published https://www.npmjs.com/package/@coinbase/wallet-sdk/v/4.0.3

Reply:
@nateReiners when I remove the patch and `createEIP1193Provider` function and just use the `coinbaseWalletProvider` I get the following error after bumping to 4.0.3 - I dont get this error on 4.0.2 without the method. 
`eth_chainId` fix is working well!
<img width="1581" alt="Screenshot 2024-05-31 at 11 11 24" src="https://github.com/blocknative/web3-onboard/assets/24457880/b0868bd8-ddb3-4b4b-ab42-099fee997ac1">